src/System.Management.Automation/namespaces/FileSystemContentStream.cs

```diff
@@ -845,6 +845,8 @@ private void CreateStreams(string filePath, string streamName, FileMode fileMode
                 {
 #if !UNIX
                     _stream = AlternateDataStreamUtilities.CreateFileStream(filePath, streamName, fileMode, fileAccess, fileShare);
+#else
+                    throw new PlatformNotSupportedException();
```

Comment:
I think these exceptions need to be explicit in indicating alternate data streams are not supported. That's why I suggested an exception at ctor time; an exception that indicates streamName is not supported on UNIX.